### ISSUE TITLE
apio 0.94 fails to install oss-cad-suite on x86_64 mac because the link is broken

### ISSUE THREAD

**Issue by adennen** [FPGAwars/apio#379]
It looks like apio is following an invalid link when trying to download oss-cad-suite on macos

this link:
https://github.com/FPGAwars/tools-oss-cad-suite/releases/download/v0.0.9/tools-oss-cad-suite-darwin_x86_64-0.0.9.tar.gz

should be changed to:
https://github.com/FPGAwars/tools-oss-cad-suite/releases/download/v0.0.9/tools-oss-cad-suite-darwin-0.0.9.tar.gz

```
apio install --all
File version.txt downloaded!
Version: 0.0.36
Installing examples package:
Already installed. Version 0.0.36
File version.txt downloaded!
Version: 0.0.9
Installing oss-cad-suite package:
Got an unrecognized status code: 404
When downloading https://github.com/FPGAwars/tools-oss-cad-suite/releases/download/v0.0.9/tools-oss-cad-suite-darwin_x86_64-0.0.9.tar.gz
Error: Package not found
```

**Comment by Obijuan** [comment]
Hi @adennen !

Could you please show me the output of this command when executed in your system?
```
apio system -i
```

@zapta , I think you are also using apio in Mac, could you please check what happens in your system when execute the `apio install --all`. Can you reproduce the bug? (It would also be great if you could provide the output of the command `apio system -l` in your mac. Thanks a lot

**Comment by zapta** [comment]
Juan, the information is below. Please let me know if you need anything else. `apio system` -l gave me an error (no such flag -l, so I skipped it). My computer is Mac with Apple M2 silicon (non x86).

Script:
```
#!/bin/bash -x

set -e

# Sys info
system_profiler SPSoftwareDataType SPHardwareDataType

pip freeze | xargs pip uninstall -y

rm -rf ~/.apio

pip freeze

pip install apio

pip install packaging

which apio

#apio system -l

apio install --all

apio clean

apio build

```

Log:
```
/Users/user/projects/fpga/repo/hdl $ ./test.sh 
+ set -e
+ system_profiler SPSoftwareDataType SPHardwareDataType
Software:

    System Software Overview:

      System Version: macOS 14.5 (23F79)
      Kernel Version: Darwin 23.5.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Computer Name: User’s MacBook Pro
      User Name: User (user)
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
      Time since boot: 2 days, 4 hours, 19 minutes

Hardware:

    Hardware Overview:

      Model Name: MacBook Pro
      Model Identifier: Mac14,7
      Chip: Apple M2
      Total Number of Cores: 8 (4 performance and 4 efficiency)
      Memory: 8 GB
      System Firmware Version: 10151.121.1
      OS Loader Version: 10151.121.1

+ pip freeze
+ xargs pip uninstall -y
Found existing installation: apio 0.9.4
Uninstalling apio-0.9.4:
  Successfully uninstalled apio-0.9.4
Found existing installation: certifi 2024.6.2
Uninstalling certifi-2024.6.2:
  Successfully uninstalled certifi-2024.6.2
Found existing installation: charset-normalizer 3.3.2
Uninstalling charset-normalizer-3.3.2:
  Successfully uninstalled charset-normalizer-3.3.2
Found existing installation: click 8.1.3
Uninstalling click-8.1.3:
  Successfully uninstalled click-8.1.3
Found existing installation: colorama 0.4.6
Uninstalling colorama-0.4.6:
  Successfully uninstalled colorama-0.4.6
Found existing installation: idna 3.7
Uninstalling idna-3.7:
  Successfully uninstalled idna-3.7
Found existing installation: packaging 24.1
Uninstalling packaging-24.1:
  Successfully uninstalled packaging-24.1
Found existing installation: pyserial 3.5
Uninstalling pyserial-3.5:
  Successfully uninstalled pyserial-3.5
Found existing installation: requests 2.28.2
Uninstalling requests-2.28.2:
  Successfully uninstalled requests-2.28.2
Found existing installation: SCons 4.2.0
Uninstalling SCons-4.2.0:
  Successfully uninstalled SCons-4.2.0
Found existing installation: semantic-version 2.9.0
Uninstalling semantic-version-2.9.0:
  Successfully uninstalled semantic-version-2.9.0
Found existing installation: setuptools 70.0.0
Uninstalling setuptools-70.0.0:
  Successfully uninstalled setuptools-70.0.0
Found existing installation: urllib3 1.26.18
Uninstalling urllib3-1.26.18:
  Successfully uninstalled urllib3-1.26.18
Found existing installation: wheel 0.43.0
Uninstalling wheel-0.43.0:
  Successfully uninstalled wheel-0.43.0
+ rm -rf /Users/user/.apio
+ pip freeze
+ pip install apio
Collecting apio
  Using cached apio-0.9.4-py3-none-any.whl.metadata (5.8 kB)
Collecting click==8.1.3 (from apio)
  Using cached click-8.1.3-py3-none-any.whl.metadata (3.2 kB)
Collecting semantic_version==2.9.0 (from apio)
  Using cached semantic_version-2.9.0-py2.py3-none-any.whl.metadata (9.7 kB)
Collecting requests==2.28.2 (from apio)
  Using cached requests-2.28.2-py3-none-any.whl.metadata (4.6 kB)
Collecting colorama==0.4.6 (from apio)
  Using cached colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)
Collecting pyserial==3.5 (from apio)
  Using cached pyserial-3.5-py2.py3-none-any.whl.metadata (1.6 kB)
Collecting wheel<1,>=0.35.0 (from apio)
  Using cached wheel-0.43.0-py3-none-any.whl.metadata (2.2 kB)
Collecting scons==4.2.0 (from apio)
  Using cached SCons-4.2.0-py3-none-any.whl.metadata (5.4 kB)
Collecting charset-normalizer<4,>=2 (from requests==2.28.2->apio)
  Using cached charset_normalizer-3.3.2-cp312-cp312-macosx_11_0_arm64.whl.metadata (33 kB)
Collecting idna<4,>=2.5 (from requests==2.28.2->apio)
  Using cached idna-3.7-py3-none-any.whl.metadata (9.9 kB)
Collecting urllib3<1.27,>=1.21.1 (from requests==2.28.2->apio)
  Using cached urllib3-1.26.18-py2.py3-none-any.whl.metadata (48 kB)
Collecting certifi>=2017.4.17 (from requests==2.28.2->apio)
  Using cached certifi-2024.6.2-py3-none-any.whl.metadata (2.2 kB)
Collecting setuptools (from scons==4.2.0->apio)
  Using cached setuptools-70.0.0-py3-none-any.whl.metadata (5.9 kB)
Using cached apio-0.9.4-py3-none-any.whl (91 kB)
Using cached click-8.1.3-py3-none-any.whl (96 kB)
Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Using cached pyserial-3.5-py2.py3-none-any.whl (90 kB)
Using cached requests-2.28.2-py3-none-any.whl (62 kB)
Using cached SCons-4.2.0-py3-none-any.whl (4.2 MB)
Using cached semantic_version-2.9.0-py2.py3-none-any.whl (15 kB)
Using cached wheel-0.43.0-py3-none-any.whl (65 kB)
Using cached certifi-2024.6.2-py3-none-any.whl (164 kB)
Using cached charset_normalizer-3.3.2-cp312-cp312-macosx_11_0_arm64.whl (119 kB)
Using cached idna-3.7-py3-none-any.whl (66 kB)
Using cached urllib3-1.26.18-py2.py3-none-any.whl (143 kB)
Using cached setuptools-70.0.0-py3-none-any.whl (863 kB)
Installing collected packages: pyserial, wheel, urllib3, setuptools, semantic_version, idna, colorama, click, charset-normalizer, certifi, scons, requests, apio
Successfully installed apio-0.9.4 certifi-2024.6.2 charset-normalizer-3.3.2 click-8.1.3 colorama-0.4.6 idna-3.7 pyserial-3.5 requests-2.28.2 scons-4.2.0 semantic_version-2.9.0 setuptools-70.0.0 urllib3-1.26.18 wheel-0.43.0
+ pip install packaging
Collecting packaging
  Using cached packaging-24.1-py3-none-any.whl.metadata (3.2 kB)
Using cached packaging-24.1-py3-none-any.whl (53 kB)
Installing collected packages: packaging
Successfully installed packaging-24.1
+ which apio
/Library/Frameworks/Python.framework/Versions/3.12/bin/apio
+ apio install --all
File version.txt downloaded!
Version: 0.0.36
Installing examples package:
Download apio-examples-0.0.36.zip
Downloading  [████████████████████████████████████]  100%
Unpacking..  [████████████████████████████████████]  100%
Package 'examples' has been successfully installed!
File version.txt downloaded!
Version: 0.0.9
Installing oss-cad-suite package:
Download tools-oss-cad-suite-darwin_arm64-0.0.9.tar.gz
Downloading  [████████████████████████████████████]  100%          
Unpacking..  [████████████████████████████████████]  100%          
Package 'oss-cad-suite' has been successfully installed!
+ apio clean
Removed hardware.json
Removed hardware.asc
Removed hardware.bin
======================================================= [SUCCESS] Took 0.08 seconds =======================================================
+ apio build
[Wed Jun 12 13:49:41 2024] Processing upduino31
-------------------------------------------------------------------------------------------------------------------------------------------
yosys -p "synth_ice40 -top main -json hardware.json" -q fifo_tx.v i2s_rx.v i2s_test_pattern.v i2s_timing.v main.v meta_stability.v queue.v queue_pusher.v register.v reset_gen.v status_leds.v
Warning: Replacing memory \arr with list of registers. See i2s_rx.v:34
nextpnr-ice40 --up5k --package sg48 --json hardware.json --asc hardware.asc --pcf main.pcf -q
icepack hardware.asc hardware.bin
======================================================= [SUCCESS] Took 5.95 seconds =======================================================
/Users/user/projects/fpga/repo/hdl $ 


```

**Comment by Obijuan** [comment]
> Juan, the information is below. Please let me know if you need anything else. apio system -l gave me an error (no such flag -l, so I skipped it). My computer is Mac with Apple M2 silicon (non x86).

ups, i made a miskate. I want you to test this command instead: ```apio system -i```

@adennen I also need you to test this command and show me the output, in order to fix this bug: ```apio system -i```
I will try to fix it for the next bug release (apio 0.9.5) in a few days

**Comment by zapta** [comment]
$ apio system -i

Platform: darwin_arm64

**Comment by Obijuan** [comment]
> $ apio system -i
Platform: darwin_arm64

Thanks for the test! There is no bug for the darwin_arm64 platform. I need @adennen to provide information about its system in order to fix the bug. If not provided I will release apio 0.9.5 stable. If the bug is confirmed we can fix it in the next release cycle

**Comment by Obijuan** [comment]
I think I've fixed it in apio stable 0.9.5. I still need @adennen to execute `apio system -i` for getting more information about the bug  
I also need @adennen to install apio 0.9.5 and execute `apio install --all` for confirming that the bug is fixed
I will keep this issue open until I got some feedback

**Comment by Obijuan** [comment]
I am closing this issue as there are no feedback. Please, feel free to reopen it if the bug remains
